contract_manager/scripts/deploy_evm_pricefeed_contracts.ts

```diff
@@ -0,0 +1,298 @@
+import { hideBin } from "yargs/helpers";
+import { EvmChain } from "../src/chains";
+import { DefaultStore } from "../src/store";
+import { existsSync, readFileSync, writeFileSync } from "fs";
+import {
+  DeploymentType,
+  EvmPriceFeedContract,
+  getDefaultDeploymentConfig,
+  PrivateKey,
+  toPrivateKey,
+  WormholeEvmContract,
+} from "../src";
+import { join } from "path";
+import Web3 from "web3";
+import { Contract } from "web3-eth-contract";
+
+type DeploymentConfig = {
+  type: DeploymentType;
+  validTimePeriodSeconds: number;
+  singleUpdateFeeInWei: number;
+  gasMultiplier: number;
+  gasPriceMultiplier: number;
+  privateKey: PrivateKey;
+  jsonOutputDir: string;
+  saveContract: boolean;
+};
+
+const CACHE_FILE = ".cache-deploy-evm";
+
+const parser = yargs(hideBin(process.argv))
+  .scriptName("deploy_evm_pricefeed_contracts.ts")
+  .usage(
+    "Usage: $0 --std-output-dir <path/to/std-output-dir/> --private-key <private-key> --chain <chain0> --chain <chain1>"
+  )
+  .options({
+    "std-output-dir": {
+      type: "string",
+      demandOption: true,
+      desc: "Path to the standard JSON output of the contracts (build artifact) directory",
+    },
+    "private-key": {
+      type: "string",
+      demandOption: true,
+      desc: "Private key to use for the deployment",
+    },
+    chain: {
+      type: "array",
+      demandOption: true,
+      desc: "Chain to upload the contract on. Can be one of the evm chains available in the store",
+    },
+    type: {
+      type: "string",
+      demandOption: false,
+      default: "stable",
+      desc: "Deployment type to use. Can be 'stable' or 'beta'",
+    },
+    "valid-time-period-seconds": {
+      type: "number",
+      demandOption: false,
+      default: 60,
+      desc: "Valid time period in seconds for the price feed staleness",
+    },
+    "single-update-fee-in-wei": {
+      type: "number",
+      demandOption: false,
+      default: 1,
+      desc: "Single update fee in wei for the price feed",
+    },
+    "gas-multiplier": {
+      type: "number",
+      demandOption: false,
+      // Pyth Proxy (ERC1967) gas estimate is insufficient in many networks and thus we use 2 by default to make it work.
+      default: 2,
+      desc: "Gas multiplier to use for the deployment. This is useful when gas estimates are not accurate",
+    },
+    "gas-price-multiplier": {
+      type: "number",
+      demandOption: false,
+      default: 1,
+      desc: "Gas price multiplier to use for the deployment. This is useful when gas price estimates are not accurate",
+    },
+    "save-contract": {
+      type: "boolean",
+      demandOption: false,
+      default: true,
+      desc: "Save the contract to the store",
+    },
+  });
+
+async function deployIfNotCached(
+  chain: EvmChain,
+  config: DeploymentConfig,
+  artifactName: string,
+  deployArgs: any[] // eslint-disable-line  @typescript-eslint/no-explicit-any
+): Promise<string> {
+  const cache = existsSync(CACHE_FILE)
```

Comment:
Can we use runIfNotCached here?

Reply:
Yeah it's a good idea. Let me do it in a separate PR (because this one is already very big)